documentation/smg/08_setting_up_jenkins_ci.md

```diff
@@ -325,3 +325,49 @@ withCredentials([string(credentialsId: 'TOKEN_ID', variable: 'secret')]) {
   """
 }
 ```
+
+## Established Jenkins pipelines
+
+### PR
+
+#### Herbert
+
+<img src="./images/08_HerbertPR.svg">
+
+#### Horace
+
+<img src="./images/08_HoracePR.svg">
+
+### Nightly
+
+#### Herbert
+
+<img src="./images/08_HerbertNightly.svg">
+
+#### Horace
+
```

Comment:
For this Horace nightly diagram, a suitable legend could be (in the text) "This diagram shows the process of triggering and running a build of Horace based on changes in the master branch following one or more pushes of a developer's branch to origin, or a successful rebuild of Herbert on the master branch, in the last 24 hours".

I'm not clear about the Jenkins timer loop. It looks as if there is an uncontrolled fork of two processes, one into Herbert Nightly and one into checking the Jenkins timer. Does this mean that the Herbert build does not fire on its own based on a Herbert PR, but is fired from Horace? Or does it just mean that Herbert is checked for a rebuild before Horace starts? I would have expected the left parallelogram to say "Check Herbert nightly rebuilt" and a second "Has changed" diamond in that branch.

Master presumably Horace master, but is shown feeding into Herbert Nightly. I would have expected it to feed into Jenkins Timer as in the above Herbert diagram.

I do not understand the rectange "Unstable" feeding into the documentation push.

Reply:
The "unstable" is the folder name for the Docs to be put in.